src/pools.ts

```diff
@@ -29,10 +38,63 @@ export class Pools
     return await this.contract.USDC()
   }
 
+  async isPaused(): Promise<boolean> {
+    return await this.contract.paused()
+  }
+
+  private _checkIsNotPaused = async () => {
+    if (await this.isPaused()) {
+      throw new Error(poolCommonErrorMessage.contractIsPaused)
+    }
+  }
+
+  private _checkIsAdmin = async () => {
+    if (this.signer) {
+      const isAdmin = await this.contract.hasRole(Role.ADMIN, this.signer)
+
+      if (!isAdmin) {
+        throw new Error(poolCommonErrorMessage.addressIsNotAdmin)
+      }
+    }
+  }
+
+  private _getStatusByIndex = (index: bigint) => {
+    const statusOptions = Object.keys(PoolStatusOption)
+    const status = Number(index)
+
+    if (status < 0 || status >= statusOptions.length) {
+      throw new Error(poolCommonErrorMessage.noSupportedPoolStatus(index))
+    }
+
+    return statusOptions[status] as PoolStatus
+  }
+
   private async _getPoolById(poolId: bigint): Promise<Pool | null> {
-    const pool: Pool = await this.contract.pools(poolId)
+    if (poolId < 0 || poolId >= (await this.contract.poolIndex())) return null
```

Comment:
Do not use await calls inside of an if statement, it makes the code harder to read